pandas/core/groupby/generic.py

```diff
@@ -1288,7 +1296,7 @@ def _wrap_applied_output(self, keys, values, not_indexed_same=False):
                         **kwargs, dtype_if_empty=object
                     )
                 else:
-                    backup = first_not_none._constructor(**kwargs)
+                    backup = first_not_none._constructor(**kwargs, dtype=object)
```

Comment:
really you had to add a dtype?

Reply:
I had added the dtype to avoid the DeprecationWarning when called with a `Series` (or subclass). You're correct that this is probably unnecessary/wrong -- I'll revert this back.

Reply:
I changed the if statement above this line to check if `isinstance(first_not_none, Series)` so that it would also work for subclassed `Series`. This avoids the DeprecationWarnings that were being raised by my tests.